src/Evolution/Systems/Cce/AnalyticSolutions/LinearizedBondiSachs.cpp

```diff
@@ -0,0 +1,511 @@
+                                      j_3_coefficient, l_max, time);
+}
+
+void LinearizedBondiSachs::linearized_dt_bondi_u(
+    const gsl::not_null<SpinWeighted<ComplexDataVector, 1>*> dt_bondi_u,
+    const size_t l_max, const double time) const noexcept {
+  const std::complex<double> u_2_coefficient =
+      sqrt(3.0) * (0.5 * c_2a_ / square(extraction_radius_) +
+                   0.25 * c_2b_ / pow<4>(extraction_radius_) +
+                   std::complex<double>(1.0, 0.0) * frequency_ *
+                       (c_2b_ / (3.0 * extraction_radius_)));
+
+  const std::complex<double> u_3_coefficient =
+      sqrt(6.0) *
+      (0.5 * c_3a_ / square(extraction_radius_) -
+       2.0 * square(frequency_) * c_3b_ / (3.0 * pow<3>(extraction_radius_)) +
+       c_3b_ / pow<5>(extraction_radius_) +
+       std::complex<double>(0.0, 1.0) * frequency_ *
+           (1.25 * c_3b_ / pow<4>(extraction_radius_)));
+
+  assign_dt_components_from_l_factors(dt_bondi_u, u_2_coefficient,
+                                      u_3_coefficient, l_max, time);
+}
+
+void LinearizedBondiSachs::linearized_dt_bondi_w(
+    const gsl::not_null<SpinWeighted<ComplexDataVector, 0>*> dt_bondi_w,
+    const size_t l_max, const double time) const noexcept {
+  const std::complex<double> w_2_coefficient =
+      -square(frequency_) * c_2b_ / square(extraction_radius_) +
+      0.5 * c_2b_ / pow<4>(extraction_radius_) +
+      std::complex<double>(0.0, 1.0) * frequency_ *
+          (c_2b_ / pow<3>(extraction_radius_));
+
+  const std::complex<double> w_3_coefficient =
+      2.5 * frequency_ * c_3b_ / pow<4>(extraction_radius_) +
+      3.0 * c_3b_ / pow<5>(extraction_radius_) +
+      std::complex<double>(0.0, 1.0) * frequency_ *
+          (-2.0 * square(frequency_) * c_3b_ / square(extraction_radius_) -
+           4.0 * frequency_ * c_3b_ / pow<3>(extraction_radius_));
+
+  assign_dt_components_from_l_factors(dt_bondi_w, w_2_coefficient,
+                                      w_3_coefficient, l_max, time);
+}
+
+void LinearizedBondiSachs::spherical_metric(
+    const gsl::not_null<
+        tnsr::aa<DataVector, 3, ::Frame::Spherical<::Frame::Inertial>>*>
+        spherical_metric,
+    const size_t l_max, const double time) const noexcept {
+  Variables<tmpl::list<Tags::BondiJ, Tags::BondiU, Tags::BondiK, Tags::BondiW>>
+      temporary_variables{
+          Spectral::Swsh::number_of_swsh_collocation_points(l_max)};
+  auto& bondi_j = get(get<Tags::BondiJ>(temporary_variables));
+  auto& bondi_u = get(get<Tags::BondiU>(temporary_variables));
+  auto& bondi_k = get(get<Tags::BondiK>(temporary_variables));
+  auto& bondi_w = get(get<Tags::BondiW>(temporary_variables));
+
+  linearized_bondi_j(make_not_null(&bondi_j), l_max, time);
+  linearized_bondi_u(make_not_null(&bondi_u), l_max, time);
+  linearized_bondi_w(make_not_null(&bondi_w), l_max, time);
+  bondi_k = sqrt(1.0 + bondi_j * conj(bondi_j));
+
+  get<0, 0>(*spherical_metric) =
```

Comment:
we're too confused to compare with the documentation...

Reply:
Oh, yeah, you're right -- I think the angular factors are probably too complicated to present as I have done. I'll just redo the documentation to explicitly list out the components.